Source/WebCore/dom/ContainerNode.h

```diff
@@ -59,8 +59,8 @@ class ContainerNode : public Node {
     void stringReplaceAll(String&&);
     void replaceAll(Node*);
 
-    ContainerNode& rootNode() const { return downcast<ContainerNode>(Node::rootNode()); }
-    Ref<ContainerNode> protectedRootNode() const { return downcast<ContainerNode>(Node::rootNode()); }
+    ContainerNode& rootNode() const { return uncheckedDowncast<ContainerNode>(Node::rootNode()); }
```

Comment:
Hm... it's not immediately obvious to me that this cast is always safe.
I get that ContainerNode can't have non-ContainerNode as its parent but maybe it's best to play safe & check that at runtime.

Reply:
Ok, I'll look into making it more obvious. I was trying to avoid it because it duplicates some code.

Reply:
Ok. Hopefully things are clearer now. I still have a tiny regression on Intel when turning on the RELEASE_ASSERT() in downcast<>() so I'm trying to be aggressive enough to turn it on.